File: 05-vetor/Exemplo1.java
/**
 * Demonstração que em Java ocorre erro de execução (ArrayIndexOutOfBoundsException) ao tentar acessar um índice inválido de um vetor.
 */
import java.util.Scanner;

public class Exemplo1 {

    public static void main(String[] args) {
        Scanner entrada = new Scanner(System.in);
        int[] vetor = new int[5];
        
        System.out.println("Digite 5 números inteiros:");
        vetor[0] = entrada.nextInt();
        vetor[1] = entrada.nextInt();
        vetor[2] = entrada.nextInt();
        vetor[3] = entrada.nextInt();
        vetor[4] = entrada.nextInt();

        System.out.println("\nOs números digitados foram:");
        System.out.println(vetor[0]);
        System.out.println(vetor[1]);
        System.out.println(vetor[2]);
        System.out.println(vetor[3]);
        System.out.println(vetor[4]);
        
        System.out.println("O vetor tem tamanho " + vetor.length);
        
        // Aqui vai ocorrer um erro de execução (ArrayIndexOutOfBoundsException)
        System.out.println(vetor[5]);
    }
}

File: 05-vetor/exemplo-1.c
# include <stdio.h>

int main(void) {
    int vetor[5];

    scanf("%d", &vetor[0]);
    scanf("%d", &vetor[1]);
    scanf("%d", &vetor[2]);
    scanf("%d", &vetor[3]);
    scanf("%d", &vetor[4]);

    printf("\nOs valores digitados foram:\n");
    printf("%d\n", vetor[0]);
    printf("%d\n", vetor[1]);
    printf("%d\n", vetor[2]);
    printf("%d\n", vetor[3]);
    printf("%d\n", vetor[4]);
    // printf("%d\n", vetor[5]); // Acessa um índice inválido


    return 0;
}